src/allmydata/test/test_storage.py

```diff
@@ -1379,6 +1404,40 @@ def secrets(n):
               {0: ([], [(500, b"make me really bigger")], None)}, [])
         self.compare_leases_without_timestamps(all_leases, list(s0.get_leases()))
 
+    def test_mutable_add_lease_renews(self):
+        """
+        Adding a lease on an already leased mutable just renews it.
```

Comment:
"... with the same lease renewal secret ..."

Reply:
I wonder if there are any tests that exercise creating *multiple* leases ... ie, passing different renew secrets in the right place?

Reply:
`test_leases` might? Feel like this branch is enough for now though.